File: my-credentials-plugin/src/main/java/holygradle/credentials/HideDialogTimerTask.java
package holygradle.credentials;

import javax.swing.*;
import java.util.TimerTask;

class HideDialogTimerTask extends TimerTask {
    private final JDialog m_dialog;
    private boolean fired = false;
    
    public HideDialogTimerTask(JDialog dialog) {
        m_dialog = dialog;
    }
    
    public void run() {
        fired = true;
        m_dialog.setVisible(false);
    }
    
    public boolean didFire() {
        return fired;
    }
}
      

File: my-credentials-plugin/src/main/kotlin/holygradle/credentials/HideDialogTimerTask.kt
package holygradle.credentials

import javax.swing.*
import java.util.TimerTask

class HideDialogTimerTask(private val dialog: JDialog) : TimerTask() {
    var fired = false
        private set
    
    override fun run() {
        fired = true
        dialog.isVisible = false
    }
}
